pkg/controller/migcluster/validation.go

```diff
@@ -129,58 +131,69 @@ func (r ReconcileMigCluster) validateSaSecret(cluster *migapi.MigCluster) error
 
 	// Not needed.
 	if cluster.Spec.IsHostCluster {
+		log.V(4).Info("skipping sa secret validation as this is a host cluster",
+			"name", cluster.Name, "namespace", cluster.Namespace)
 		return nil
 	}
 
 	// NotSet
 	if !migref.RefSet(ref) {
-		cluster.Status.SetCondition(migapi.Condition{
+		cluster.Status.SetAndLogCondition(log, migapi.Condition{
 			Type:     InvalidSaSecretRef,
 			Status:   True,
 			Reason:   NotSet,
 			Category: Critical,
-			Message:  InvalidSaSecretRefMessage,
-		})
+			Message:  "serviceAccountSecretRef not set as a secret for the migcluster",
+		}, cluster.Namespace, cluster.Name)
 		return nil
 	}
 
 	secret, err := migapi.GetSecret(r, ref)
 	if err != nil {
-		return liberr.Wrap(err)
+		return liberr.Wrap(fmt.Errorf("error getting migcluster sa secret %s/%s: %v",
+			cluster.Spec.ServiceAccountSecretRef.Namespace,
+			cluster.Spec.ServiceAccountSecretRef.Name,
+			err))
 	}
 
 	// NotFound
 	if secret == nil {
-		cluster.Status.SetCondition(migapi.Condition{
+		cluster.Status.SetAndLogCondition(log, migapi.Condition{
 			Type:     InvalidSaSecretRef,
 			Status:   True,
 			Reason:   NotFound,
 			Category: Critical,
-			Message:  InvalidSaSecretRefMessage,
-		})
+			Message: fmt.Sprintf("couldn't find sa secret for migcluster %s/%s",
+				cluster.Spec.ServiceAccountSecretRef.Namespace,
+				cluster.Spec.ServiceAccountSecretRef.Name),
+		}, cluster.Namespace, cluster.Name)
 		return nil
 	}
 
 	// saToken
 	token, found := secret.Data[migapi.SaToken]
 	if !found {
-		cluster.Status.SetCondition(migapi.Condition{
+		cluster.Status.SetAndLogCondition(log, migapi.Condition{
 			Type:     InvalidSaToken,
 			Status:   True,
 			Reason:   NotFound,
 			Category: Critical,
-			Message:  InvalidSaTokenMessage,
-		})
+			Message: fmt.Sprintf("couldn't find sa secret token for migcluster %s/%s",
```

Comment:
wonder if this will be a better description? `"couldn't find 'saToken' key in secret %s/%s for migcluster"`

Reply:
Done